## Detailed description of score options

| Flag         | Stage   | Description                                                                                                                                                                                                                   |
|-------------|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| flag        | Profile | Skips reads that have a SAM flag set. Default [0], no filter.                                                                                                                                                                 |
| min-mapq    | Profile | Skip reads that have a mapping quality below threshold. Default [10]. Tends to remove reads in large segmental duplications, centromeres, and telomeres. A parameter sweep from MQ5â€“MQ20 had <0.1% impact on the number of high markers. |
| min-mg      | Profile | Minimum gap compressed sequence identity (see pbmm2). Default [98%]. Typically removes reads that are paralogs or poorly mapping.                                                                                             |
| max-filt-frac | Profile | The fraction of reads that fail filters within a haplotype group; if >N% reads within a haplotype group at a marker site fail filters, a haplotype is not called for this group. Default [0.2].                                 |
| min-cov     | Profile | Minimum coverage per haplotype. Default [5]. Haplotypes falling below the threshold are not called.                                                                                                                           |
| min-bq      | Profile | Minimum read base quality value for repeat region. Default [25].                                                                                                                                                              |
| min-idt     | Profile | The minimum identity for the repeat alignment. Default [90]. Reads with a motif alignment score falling below the threshold are skipped.                                                                                      |
| min-depth   | Score   | Minimum coverage per haplotype. Default [5]. Haplotypes falling below the threshold are skipped.                                                                                                                             |
| unphased    | Score   | Exclude unphased reads (haplotype 0).                                                                                                                                                                                         |
| min-pass    | Score   | The minimum percentage of sites failing filters (not called) before a sample is labeled as failing QC. Default [50%].                                                                                                        |
